### PR TITLE
Initial...second commit for Tilemap and basic AOE

### PR DESCRIPTION
**Pull request by stephan1994w**
Pls

### CODE REVIEW

File: Assets/KriptoFX/Realistic Effects Pack v4/Scripts/Curves/RFX4_ShaderColorGradient.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    void OnDisable()
+    {
+        if (mat == null) return;
+        if (UseSharedMaterial) mat.SetColor(propertyID, startColor);
+        mat.SetColor(propertyID, startColor);
+    }
+
+    //void OnDestroy()
```
Comment (RyanGallagher859) [inline Assets/KriptoFX/Realistic Effects Pack v4/Scripts/Curves/RFX4_ShaderColorGradient.cs:103]: If this is to be used at some point add TODO else gut

File: Assets/KriptoFX/Realistic Effects Pack v4/Scripts/Editor/RFX4_CustomMaterialInspectorParticle.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+public class RFX4_CustomMaterialInspectorParticle : MaterialEditor
+{
+    private int blendIndex;
+    private int srcMode = 1;
+    private int dstMode = 1;
+    private string[] blendMode = { "Additive", "Alpha Blend", "Mul", "Mul2" };
+    string[] lightOptions = { "Vert OFF", "Vert ON", "Vert Normal ON"};
+    string[] lightOptionsKeyWords = { "VertLight_OFF", "VertLight4_ON", "VertLight4Normal_ON"};
+    private int lightIndex;
+    //private Vector2 tiling = new Vector2(800, 800);
```
Comment (RyanGallagher859) [inline Assets/KriptoFX/Realistic Effects Pack v4/Scripts/Editor/RFX4_CustomMaterialInspectorParticle.cs:15]: remove

File: Assets/KriptoFX/Realistic Effects Pack v4/Scripts/Editor/RFX4_CustomMaterialInspectorParticle.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        var shaderKeyWords = targetMat.shaderKeywords;
+       
+        foreach (var keyWord in shaderKeyWords)
+        {
+            var index = lightOptionsKeyWords.ToList().IndexOf(keyWord);
+            if (index >= 0)
+                lightIndex = index;
+        }
+        var tiling4 = targetMat.GetVector("_Tiling");
+        //tiling = new Vector2(tiling4.x, tiling4.y);
```
Comment (RyanGallagher859) [inline Assets/KriptoFX/Realistic Effects Pack v4/Scripts/Editor/RFX4_CustomMaterialInspectorParticle.cs:40]: remove

File: Assets/Scripts/Player.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+    public float speedMultiplier;
+    public float sprintMultiplier;
+    public float jumpMultiplier;
+    public float jumpCooldown;
+    public float throwForce = 45f;
+
+    public Crosshairs crosshairs;
+    public GameObject grenadePrefab;
+    public GameObject AreaOfEffectType;
```
Comment (RyanGallagher859) [inline Assets/Scripts/Player.cs]: Both grenade and areaOfEffectType may not be used per character. These would be added in the prefab stage of character building right?

File: Assets/Scripts/Player.cs [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        }
+
+        if (Input.GetMouseButtonDown(1))
+        {
+            AOE();
+        }
+
+    }
+
+    void ThrowGrenade()
```
Comment (RyanGallagher859) [inline Assets/Scripts/Player.cs]: Id have a grenade script, what you think?

File: Assets/Scripts/Player.cs [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        Ray ray = viewCamera.ScreenPointToRay(Input.mousePosition);
+        Plane groundPlane = new Plane(Vector3.back, Vector3.zero);
+        float rayDistance;
+
+        if (groundPlane.Raycast(ray, out rayDistance))
+        {
+            Vector2 point = ray.GetPoint(rayDistance);
+            crosshairs.transform.position = point;
+            //crosshairs.DetectTargets(ray);
+            //Debug.DrawRay(transform.position, forward, Color.green);
```
Comment (RyanGallagher859) [inline Assets/Scripts/Player.cs]: Remove unused comments

File: Assets/Scripts/Player.cs [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        var playerPos = rigidBody.position;
+        Vector3 cameraPosition = Camera.main.ScreenToWorldPoint(Input.mousePosition);
+
+        var direction = new Vector2(cameraPosition.x - playerPos.x, cameraPosition.y - playerPos.y);
+        angle = Mathf.Atan2(direction.y, direction.x) * Mathf.Rad2Deg;
+
+        rigidBody.rotation = angle;
+
+        //Debug.Log(angle);
+        //Debug.DrawLine(new Vector3(pos.x, pos.y, -1), new Vector3(cameraPosition.x, cameraPosition.y, -1), Color.red, 2.5f);
```
Comment (RyanGallagher859) [inline Assets/Scripts/Player.cs]: remove

File: Assets/Scripts/AreaOfEffect.cs [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+                foreach (Collider2D nearbyObj in colliders)
+                {
+                    // Add force
+                    Rigidbody2D rb = nearbyObj.GetComponent<Rigidbody2D>();
+                    if (rb != null)
+                    {
+                        rb.AddExplosionForce(force, transform.position, radius);
+                    }
+                }
+            }
```
Comment (mshaw11) [inline Assets/Scripts/AreaOfEffect.cs:73]: Comment same as below "//explosion effect" - give comment on the aoetype == EffectType line at the top and explain why they're separate and the difference in logic.

**Review by RyanGallagher859 — APPROVED** [review]

**Pull request merged**